packages/react-utilities/src/hooks/useOnClickOutside.ts

```diff
@@ -67,7 +67,7 @@ export const useOnClickOutside = (options: UseOnClickOrScrollOutsideOptions) =>
     }
 
     // Garbage collect this event after it's no longer useful to avoid memory leaks
-    timeoutId.current = setTimeout(() => {
+    timeoutId.current = window.setTimeout(() => {
```

Comment:
without being explicit regarding from what global context are we accessing props TS will fail (basically local api-extractor build for react-menu was broken because of this)

Reply:
please use the `canUseDom` utility, otherwise this will crash during SSR.

cc @layershifter

Reply:
do we need to use that in this PR ? there are multiple other occurrences in other projects which are using `window`

> canUseDom
where does it live ?

Reply:
kudos to @layershifter for looking into this specifically and finding out that at least for `Next.js` `window` and `document` usages seem to not throw (??!!??)

It might still be better to be explicit about the SSR case. the utility function I mentioned in the same package: 

https://github.com/microsoft/fluentui/blob/master/packages/react-utilities/src/ssr/canUseDOM.ts

Reply:
I just realised, all of this is happening within `useEffect` which AFAIR is not executed when SSR, thus I'd leave it as is and do the necessary changes in other PR. there are other uses of `window` directly within this file.

![2021-06-02 at 12 31](https://user-images.githubusercontent.com/1223799/120465712-8cd5c100-c39e-11eb-8512-071ddd695df9.png)


Are we ok with this ? thx